x-pack/plugins/cases/server/common/models/case_with_comments.ts

```diff
@@ -322,4 +348,54 @@ export class CaseCommentModel {
       });
     }
   }
+  public async bulkCreate({
+    attachments,
+  }: {
+    attachments: Array<{ id: string } & CommentRequest>;
+  }): Promise<CaseCommentModel> {
+    try {
+      this.validateCreateCommentRequest(attachments);
+
+      const caseReference = this.buildRefsToCase();
+
+      const [newlyCreatedAttachments, commentableCase] = await Promise.all([
+        this.params.attachmentService.bulkCreate({
+          unsecuredSavedObjectsClient: this.params.unsecuredSavedObjectsClient,
+          attachments: attachments.map(({ id, ...attachment }) => {
+            return {
+              attributes: transformNewComment({
+                createdDate: new Date().toISOString(),
+                ...attachment,
+                ...this.params.user,
+              }),
+              references: [...caseReference, ...this.getCommentReferences(attachment)],
+              id,
+            };
+          }),
+        }),
+        this.updateCaseUserAndDate(new Date().toISOString()),
+      ]);
+
+      const savedObjectsWithoutErrors = newlyCreatedAttachments.saved_objects.filter(
+        (attachment) => attachment.error == null
+      );
+
+      const attachmentsWithoutErrors = attachments.filter((attachment) =>
+        savedObjectsWithoutErrors.some((so) => so.id === attachment.id)
+      );
+
+      await Promise.all([
```

Comment:
@cnasikas this is one of those cases I mentioned where things can fail but still succeed. e.g. the attachments are created but the user actions are not.

no need to take action, just a good example of it.